### PR TITLE
Send a proper error message for incorrect POSTs.

### PR DESCRIPTION
**Pull request by manuranga**

### CODE REVIEW

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java [hunk 1]
```diff
@@ -140,12 +140,24 @@ public String renderPage(HttpRequest request, HttpResponse response) {
                     uriWithoutContextPath.substring(0, uriWithoutContextPath.length() - 1) :
                     (uriWithoutContextPath + "/");
             if (hasPage(correctedUriWithoutContextPath)) {
-                // Redirecting to the correct page.
-                String correctedUri = request.getContextPath() + correctedUriWithoutContextPath;
-                if (request.getQueryString() != null) {
-                    correctedUri = correctedUri + '?' + request.getQueryString();
+                if (request.getMethod().equals("GET")) {
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: we can use `request.isGetRequest()` method here

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java [hunk 2]
```diff
@@ -140,12 +140,24 @@ public String renderPage(HttpRequest request, HttpResponse response) {
...
+                    // Redirecting to the correct page.
+                    String correctedUri = request.getContextPath() + correctedUriWithoutContextPath;
+                    if (request.getQueryString() != null) {
+                        correctedUri = correctedUri + '?' + request.getQueryString();
+                    }
+                    throw new PageRedirectException(correctedUri, e);
+                } else {
+                    // If GET, we correct, since this can be an end-user error. But if POST it's the responsibility of
+                    // the dev to use correct URL. Because HTTP POST is not well supported. See :
+                    // https://softwareengineering.stackexchange.com/questions/99894/why-doesnt-http-have-post-redirect
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: Lets change this line to `See https://softwareengineering.stackexchange.com/q/99894`, shorter URL

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java [hunk 3]
```diff
@@ -140,12 +140,24 @@ public String renderPage(HttpRequest request, HttpResponse response) {
...
+                    return renderErrorPage(new PageNotFoundException(
```
Comment (this) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: Code formatting. How about moving the exception creation into a separate line?

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java (hunk 1 above)
Comment (manuranga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: +1 will change

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java (hunk 2 above)
Comment (manuranga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: +1 will change

File: components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java (hunk 3 above)
Comment (manuranga) [inline components/uuf-core/src/main/java/org/wso2/carbon/uuf/core/App.java]: Putting an Exception to a var seems bit odd to me. So I moved the error message to a variable, now it looks better.

**Pull request merged**